packages/graphql/lib/decorators/resolver.decorator.ts

```diff
@@ -20,10 +20,8 @@ function getObjectOrInterfaceTypeNameIfExists(
   nameOrType: Function,
 ): string | undefined {
   const ctor = getClassOrUndefined(nameOrType);
-  const objectTypesMetadata = TypeMetadataStorage.getObjectTypesMetadata();
-  const objectMetadata = objectTypesMetadata.find(
-    (type) => type.target === ctor,
-  );
+  const objectMetadata =
+    TypeMetadataStorage.getObjectTypeMetadataByTarget(ctor);
   if (!objectMetadata) {
     const interfaceTypesMetadata = TypeMetadataStorage.getInterfacesMetadata();
     const interfaceMetadata = interfaceTypesMetadata.find(
```

Comment:
Shouldn't we use `getInterfaceMetadataByTarget()` here as well?

Reply:
Definitley. Fixing